modules/imgproc/src/accum.simd.hpp

```diff
@@ -2624,6 +2624,60 @@ void accW_simd_(const uchar* src, float* dst, const uchar* mask, int len, int cn
             v_dst10 = v_fma(v_dst10, v_beta, v_cvt_f32(v_reinterpret_as_s32(v_src10)) * v_alpha);
             v_dst11 = v_fma(v_dst11, v_beta, v_cvt_f32(v_reinterpret_as_s32(v_src11)) * v_alpha);
 
+            v_store(dst + x, v_dst00);
+            v_store(dst + x + step, v_dst01);
+            v_store(dst + x + step * 2, v_dst10);
+            v_store(dst + x + step * 3, v_dst11);
+        }
+    } else if ( cn == 1 ){
+        //#include <stdio.h>
+        const v_float32 v_zero = vx_setall_f32((float)0);
+        int size = len * cn;
+        int y = -cVectorWidth;
+        int cc = 0;
+        for (; x <= size - cVectorWidth; x += cVectorWidth, cc = (cc+1) % cn)
+        {
+            if( cc == 0 ) y += cVectorWidth;
+
+            v_uint8 v_src = vx_load(src + x);
+            v_uint8 v_mask = vx_load(mask + y);
+
+            v_uint16 v_m0, v_m1;
+            v_expand(v_mask, v_m0, v_m1);
+            v_uint32 v_m00, v_m01, v_m10, v_m11;
+            v_expand(v_m0, v_m00, v_m01);
+            v_expand(v_m1, v_m10, v_m11);
+
+            v_float32 v_mf00, v_mf01, v_mf10, v_mf11;
+            v_mf00 = v_cvt_f32(v_reinterpret_as_s32(v_m00));
+            v_mf01 = v_cvt_f32(v_reinterpret_as_s32(v_m01));
+            v_mf10 = v_cvt_f32(v_reinterpret_as_s32(v_m10));
+            v_mf11 = v_cvt_f32(v_reinterpret_as_s32(v_m11));
+
+            v_float32 d0 = v_mf00 != v_zero;
+            v_float32 d1 = v_mf01 != v_zero;
+            v_float32 d2 = v_mf10 != v_zero;
+            v_float32 d3 = v_mf11 != v_zero;
+
+            //for(int i = 0; i < v_uint)
+
+            v_uint16 v_src0, v_src1;
+            v_expand(v_src, v_src0, v_src1);
+
+            v_uint32 v_src00, v_src01, v_src10, v_src11;
+            v_expand(v_src0, v_src00, v_src01);
+            v_expand(v_src1, v_src10, v_src11);
+
+            v_float32 v_dst00 = vx_load(dst + x);
+            v_float32 v_dst01 = vx_load(dst + x + step);
+            v_float32 v_dst10 = vx_load(dst + x + step * 2);
+            v_float32 v_dst11 = vx_load(dst + x + step * 3);
+
+            v_dst00 = v_fma(v_fma(v_dst00, v_beta, v_cvt_f32(v_reinterpret_as_s32(v_src00)) * v_alpha), d0, (~d0)*v_mf00);
```

Comment:
Looks like for zero mask this line will produce result equal to `(~d0)*v_mf00` that is definitely unrelated to expected initial dst value

Reply:
Yeah I'm fixing that in a bit, thanks for the heads up.

Reply:
It's possible to use `v_select(d0, v_fma(...), v_dst00)` intrinsic to choose between updated and original destination values